### PR TITLE
Clean up re-run logic

### PR DESCRIPTION
**Pull request by kyocum**
1.) Removed dead code
2.) Use lineage to detect when upstream data has been modified

### CODE REVIEW

**Comment by barryhicks** [comment]
I've confirmed that all the examples still run when this new branch is checked out.

File: disdat/apply.py [hunk 1]
```diff
@@ -314,13 +314,15 @@ def resolve_bundle(pfs, pipe, is_left_edge_task):
         pfs.new_output_hframe(pipe, is_left_edge_task)
 
     # 4.) Check the inputs -- assumes we have processed upstream tasks already
-    for task in pipe.requires():
+    for i, task in enumerate(pipe.requires()):
```
Comment (tmwong2003) [inline disdat/apply.py]: Is there some additional ordering that is added by using `enumerate`?

File: disdat/apply.py [hunk 2]
```diff
@@ -333,39 +335,21 @@ def resolve_bundle(pfs, pipe, is_left_edge_task):
...
+            assert(local_bundle is not None)
+
+            """ Now we need to check if the one we have cached is an old one
+            POLICY
+            1.) if the date is more recent, it is "new" data.
+            2.) if it is older, we should require force (but currently do not and re-run).
+            XXX TODO: Add date to the depends_on pb data structure to enforce 2 XXX
+            """
+            for tup in lng.pb.depends_on:
+                if tup.hframe_name == local_bundle.pb.processing_name and tup.hframe_uuid != local_bundle.pb.uuid:
```
[the remark this answers is not in the document]
Comment (kyocum) [reply disdat/apply.py]: Yes, it is possible that you would find that there is an upstream data bundle that was made with the same parameters with a different UUID that is actually older.    In this case we will re-run the pipeline.   Is this wrong?    Well, one might pull an older bundle in, and want to re-run the pipeline with the older data.    The right thing is to have enough information to tell the user what is happening.   

In this case, we should extend the depend_upon to contain dates as well.  They could also contain the lineage messages themselves, though we'd need to ensure they are shallow copies.

File: disdat/apply.py (hunk 1 above)
Comment (kyocum) [reply disdat/apply.py]: Yup, no reason.  Deleted.

**Pull request merged**